### PR TITLE
[ADP-2957] Remove encode decode stakeaddress class

### PR DESCRIPTION
**Pull request by paolino**
- [x] create new type `ApiRewardAccount n` in lieu of `(ApiT RewardAccount, Proxy n)`
- [x]  remove overlapping json instances for the type  `ApiRewardAccount n`: all golden tests involving stake_address now reports valid stake addresses
- [x] move json instances for `RewardAccount` out  of sqlite code ... :shrug: 
- [x] remove `EncodeStakeAddress n` class and use function over network singleton instead
- [x] remove `DecodeStakeAddress n` class and use function over network singleton instead
- [x] remove unneeded json golden files

ADP-2957

### CODE REVIEW

File: lib/wallet/api/http/Cardano/Wallet/Api/Types/Certificate.hs [hunk 1]
```diff
@@ -78,22 +84,40 @@ import qualified Cardano.Wallet.Primitive.Types.RewardAccount as W
 import qualified Data.Aeson.Types as Aeson
 import qualified Data.List.NonEmpty as NE
 
+newtype ApiRewardAccount (n :: NetworkDiscriminant)
+        = ApiRewardAccount W.RewardAccount
+    deriving (Eq, Generic, Show)
```
Comment (Unisay) [inline lib/wallet/api/http/Cardano/Wallet/Api/Types/Certificate.hs]:
```suggestion
newtype ApiRewardAccount (n :: NetworkDiscriminant)
    = ApiRewardAccount W.RewardAccount
    deriving stock (Eq, Generic, Show)
```

File: lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/CoinSelections.hs [hunk 2]
```diff
@@ -37,6 +36,7 @@ import Cardano.Wallet.Primitive.Types.TokenMap
     ( AssetId (..) )
 import Cardano.Wallet.Primitive.Types.TokenPolicy
     ( TokenName (..), TokenPolicyId (..) )
+
```
Comment (HeinrichApfelmus) [inline lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/CoinSelections.hs]:
```suggestion
```

Accidental import list split.

File: lib/wallet/api/http/Cardano/Wallet/Shelley.hs [hunk 3]
```diff
@@ -304,8 +300,6 @@ serveWallet
             ( PaymentAddress n IcarusKey 'CredFromKeyK
             , PaymentAddress n ByronKey 'CredFromKeyK
             , DelegationAddress n ShelleyKey 'CredFromKeyK
-            , EncodeStakeAddress n
-            , DecodeStakeAddress n
```
Comment (HeinrichApfelmus) [inline lib/wallet/api/http/Cardano/Wallet/Shelley.hs:308]: 🥳

File: lib/wallet/api/http/Cardano/Wallet/Api/Types/Primitive.hs [hunk 4]
```diff
@@ -347,20 +343,6 @@ instance ToJSON (ApiT TxScriptValidity) where
...
-    => FromJSON (ApiT RewardAccount, Proxy n)
-  where
-    parseJSON x = do
-        let proxy = Proxy @n
-        acct <- parseJSON x >>= eitherToParser
-            . bimap ShowFmt ApiT
-            . decodeStakeAddress @n
-        return (acct, proxy)
-
-instance {-# OVERLAPS #-} EncodeStakeAddress n
```
Comment (HeinrichApfelmus) [inline lib/wallet/api/http/Cardano/Wallet/Api/Types/Primitive.hs:360]: 🥳

**Review by Unisay — APPROVED** [review]

**Review by HeinrichApfelmus — APPROVED** [review]
Excellent, thank you! 😊 Minor newline accident on one import list.

**Comment by paolino** [comment]
bors +r

**Pull request merged**